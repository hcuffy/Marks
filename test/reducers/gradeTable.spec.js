import applyGradeData from '../../app/reducers/gradeTableReducer'
import {
	DISPLAY_EXAM_TABLE,
	OPEN_CLASS_LIST,
	UPDATE_EXAM_TABLE
} from '../../app/constants/actionTypes'

const initialLoadState = {
	classroom: null,
	subjectName: null,
	classroomDropdown: false,
	subDrop: false
}

const subjectData = {
	subjectName: 'Biology',
	classroom: 'Class 2',
	subjectId: 'dh667zZh748JnNn'
}

const exams = [
	{
		title: 'BIO EXAM',
		subjectId: 'dh667zZh748JnNn',
		date: '11/11/2019',
		weight: 1,
		_id: 'ueeNxOkl32Ry2kGF',
		createdAt: { $$date: 1543663446602 },
		updatedAt: { $$date: 1543663446602 }
	}
]

const grades = [
	{
		grade: 3,
		examId: 'ueeNxOkl32Ry2kGF',
		studentId: 'kkhNH7HNFDCvR5R',
		date: '11/11/2019',
		weight: 1,
		_id: 'zupNxOkl87RyKLMJ',
		createdAt: { $$date: 1543663876602 },
		updatedAt: { $$date: 1543663876602 }
	}
]

describe('test grade table reducer', () => {
	it('should use initial state', () => {
		expect(applyGradeData(initialLoadState, {})).toMatchSnapshot()
	})

	it('should handle DISPLAY_EXAM_TABLE', () => {
		expect(
			applyGradeData(
				{ ...initialLoadState },
				{ type: DISPLAY_EXAM_TABLE, payload: { subjectData, exams, grades } }
			)
		).toMatchSnapshot()
	})

	it('should handle OPEN_CLASS_LIST', () => {
		expect(
			applyGradeData(
				{ ...initialLoadState },
				{ type: OPEN_CLASS_LIST, payload: subjectData.classroom }
			)
		).toMatchSnapshot()
	})

	it('should handle UPDATE_EXAM_TABLE', () => {
		expect(
			applyGradeData(
				{ ...initialLoadState },
				{ type: UPDATE_EXAM_TABLE, payload: { exams, grades } }
			)
		).toMatchSnapshot()
	})
})
