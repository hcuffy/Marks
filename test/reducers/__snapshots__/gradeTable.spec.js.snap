// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test grade table reducer should handle DISPLAY_EXAM_TABLE 1`] = `
Object {
  "classroom": null,
  "classroomDropdown": false,
  "exams": Array [
    Object {
      "_id": "ueeNxOkl32Ry2kGF",
      "createdAt": Object {
        "$$date": 1543663446602,
      },
      "date": "11/11/2019",
      "subjectId": "dh667zZh748JnNn",
      "title": "BIO EXAM",
      "updatedAt": Object {
        "$$date": 1543663446602,
      },
      "weight": 1,
    },
  ],
  "grades": Array [
    Object {
      "_id": "zupNxOkl87RyKLMJ",
      "createdAt": Object {
        "$$date": 1543663876602,
      },
      "date": "11/11/2019",
      "examId": "ueeNxOkl32Ry2kGF",
      "grade": 3,
      "studentId": "kkhNH7HNFDCvR5R",
      "updatedAt": Object {
        "$$date": 1543663876602,
      },
      "weight": 1,
    },
  ],
  "subDrop": true,
  "subjectId": "dh667zZh748JnNn",
  "subjectName": "Biology",
}
`;

exports[`test grade table reducer should handle OPEN_CLASS_LIST 1`] = `
Object {
  "classroom": "Class 2",
  "classroomDropdown": true,
  "subDrop": false,
  "subjectName": null,
}
`;

exports[`test grade table reducer should handle UPDATE_EXAM_TABLE 1`] = `
Object {
  "classroom": null,
  "classroomDropdown": false,
  "exams": Array [
    Object {
      "_id": "ueeNxOkl32Ry2kGF",
      "createdAt": Object {
        "$$date": 1543663446602,
      },
      "date": "11/11/2019",
      "subjectId": "dh667zZh748JnNn",
      "title": "BIO EXAM",
      "updatedAt": Object {
        "$$date": 1543663446602,
      },
      "weight": 1,
    },
  ],
  "grades": Array [
    Object {
      "_id": "zupNxOkl87RyKLMJ",
      "createdAt": Object {
        "$$date": 1543663876602,
      },
      "date": "11/11/2019",
      "examId": "ueeNxOkl32Ry2kGF",
      "grade": 3,
      "studentId": "kkhNH7HNFDCvR5R",
      "updatedAt": Object {
        "$$date": 1543663876602,
      },
      "weight": 1,
    },
  ],
  "subDrop": false,
  "subjectName": null,
}
`;

exports[`test grade table reducer should use initial state 1`] = `
Object {
  "classroom": null,
  "classroomDropdown": false,
  "subDrop": false,
  "subjectName": null,
}
`;
